packages/roosterjs-content-model-plugins/lib/tableEdit/editors/features/TableMover.ts

```diff
@@ -129,3 +157,212 @@ function isTableTopVisible(editor: IEditor, rect: Rect | null, contentDiv?: Node
+        : setTableMoverCursor(editor, true, 'move');
+
+    const pos = getNodePositionFromEvent(editor, event.clientX, event.clientY);
+    if (pos) {
+        const range = editor.getDocument().createRange();
+        range.setStart(pos.node, pos.offset);
+        range.collapse(true);
+
+        editor.setDOMSelection({ type: 'range', range, isReverted: false });
+        return true;
+    }
+    return false;
+}
+
+function onDragEnd(
+    context: TableMoverContext,
+    event: MouseEvent,
+    initValue: TableMoverInitValue | undefined
+) {
+    const { editor, table, onFinishDragging: selectWholeTable } = context;
+    const element = event.target;
+    // Remove table outline rectangle
+    initValue?.tableRect.remove();
+
+    // Reset cursor
+    setTableMoverCursor(editor, false);
+
+    if (element == context.div) {
+        // Table mover was only clicked, select whole table
+        selectWholeTable(table);
+        context.onEnd();
+        return true;
+    } else {
+        // Check if table was dragged on itself or Check if element is not in editor
+        if (
+            table.contains(element as Node) ||
+            !editor.getDOMHelper().isNodeInEditor(element as Node)
+        ) {
+            editor.setDOMSelection(initValue?.initialSelection ?? null);
+            context.onEnd();
+            return false;
+        }
+
+        let insertionSuccess: boolean = false;
+
+        // Get position to insert table
+        const insertPosition = getNodePositionFromEvent(editor, event.clientX, event.clientY);
+        if (insertPosition) {
+            // Move table to new position
+            formatInsertPointWithContentModel(
+                editor,
+                insertPosition,
+                (model, context, ip) => {
+                    // Remove old table if not copying
+                    if (!(event.ctrlKey || event.metaKey)) {
+                        const [oldTable, path] = getFirstSelectedTable(model);
+                        if (oldTable) {
+                            const index = path[0].blocks.indexOf(oldTable);
+                            path[0].blocks.splice(index, 1);
+                        }
+                    }
+                    if (ip && initValue?.cmTable) {
+                        // Guarantee the cm Insertion point is selected
+                        ip.marker.isSelected = true;
+
+                        // Insert new table
+                        const doc = createContentModelDocument();
+                        doc.blocks.push(initValue.cmTable);
+                        insertionSuccess = !!mergeModel(model, doc, context, {
+                            mergeFormat: 'none',
+                            insertPosition: ip,
+                        });
+
+                        if (insertionSuccess) {
+                            // After mergeModel, the new table should be selected
+                            const [finalTable] = getFirstSelectedTable(model);
```

Comment:
The final table will be the same javascript object with initValue.cmTable, so you can use it directly here.

Reply:
I tried to make the change but it was not working while testing.

Reply:
It is ok not change here, but I'd like to understand more about the issue, since it may expose some underlying issue. Can you double check if the table is the same object in content model? If not, why?

Reply:
Any update here? Can you confirm if the cmTable object is the same one after merge?